src/shared-components/offClickWrapper/index.js

```diff
@@ -23,10 +27,19 @@ class OffClickWrapper extends React.Component {
 
   addOffClickListener = () => {
     document.addEventListener('click', this.handleOffClick, false);
+    document.addEventListener('keydown', this.handleKeyPress, false);
   };
 
   removeOffClickListener = () => {
     document.removeEventListener('click', this.handleOffClick, false);
+    document.removeEventListener('keydown', this.handleKeyPress, false);
```

Comment:
This is used in a couple of places, including in the `ImmersiveModal` to handle outside clicks. I can't think of a reason not to also respond to `Esc` for those, but I may need to also do a quick search in the PocketDerm repo to see if we are duplicating this logic there.